packages/eslint-plugin/tests/rules/consistent-type-imports.test.ts

```diff
@@ -1833,5 +1920,277 @@ const b = B;
+        let bar: B;
+      `,
+      output: `
+        import { type A, type B } from 'foo';
+        let foo: A;
+        let bar: B;
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'typeOverValue',
+          line: 2,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+        import { A, B } from 'foo';
+
+        let foo: A;
+        B();
+      `,
+      output: `
+        import { type A, B } from 'foo';
+
+        let foo: A;
+        B();
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'aImportIsOnlyTypes',
+          line: 2,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+        import { A, B } from 'foo';
+        type T = A;
+        B();
+      `,
+      output: `
+        import { type A, B } from 'foo';
+        type T = A;
+        B();
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'aImportIsOnlyTypes',
+          line: 2,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+        import { A } from 'foo';
+        import { B } from 'foo';
+        type T = A;
+        type U = B;
+      `,
+      output: `
+        import { type A } from 'foo';
+        import { type B } from 'foo';
+        type T = A;
+        type U = B;
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'typeOverValue',
+          line: 2,
+          column: 9,
+        },
+        {
+          messageId: 'typeOverValue',
+          line: 3,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+        import { A } from 'foo';
+        import B from 'foo';
+        type T = A;
+        type U = B;
+      `,
+      output: `
+        import { type A } from 'foo';
+        import type B from 'foo';
+        type T = A;
+        type U = B;
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'typeOverValue',
+          line: 2,
+          column: 9,
+        },
+        {
+          messageId: 'typeOverValue',
+          line: 3,
+          column: 9,
+        },
+      ],
+    },
+    {
+      code: `
+import A, { B, C } from 'foo';
+type T = B;
+type U = C;
+A();
+      `,
+      output: `
+import A, { type B, type C } from 'foo';
+type T = B;
+type U = C;
+A();
+      `,
+      options: [{ prefer: 'type-imports', fixStyle: 'inline-type-imports' }],
+      errors: [
+        {
+          messageId: 'someImportsAreOnlyTypes',
+          line: 2,
+          column: 1,
+        },
+      ],
+    },
+    {
+      code: `
+import A, { B, C } from 'foo';
+type T = B;
+type U = C;
+type V = A;
+      `,
+      output: noFormat`
```

Comment:
These actually aren't necessary anymore:

```suggestion
      output: `
```